code/modules/reagents/machinery/chem_master.dm

```diff
@@ -136,418 +136,468 @@ var/global/list/pillIcon2Name = list("oblong purple-pink", "oblong green-white",
-				dat += "<tr>"
-				dat += {"
-					<td class="column1">
-						[G.name] , [round(G.volume, 0.01)] Units - <A href='?src=\ref[src];analyze=\ref[G]'>(?)</A>
-					</td>
-					<td class="column2">
-						<A href='?src=\ref[src];add=[G.id];amount=1'>1u</A>
-						<A href='?src=\ref[src];add=[G.id];amount=5'>5u</A>
-						<A href='?src=\ref[src];add=[G.id];amount=10'>10u</A>
-						<A href='?src=\ref[src];addcustom=[G.id]'>Custom</A>
-						<A href='?src=\ref[src];add=[G.id];amount=[G.volume]'>All</A>
-					</td>
-				"}
-				dat += "</tr>"
-			dat += "</table>"
-
-		//
-		// BUFFER
-		//
-
-		dat += "<HR>"
-		if(reagents.total_volume)
-			dat += {"
-				<table>
-					<td class="column1">
-						Transfer to <A href='?src=\ref[src];toggle=1'>[(!mode ? "disposal" : "beaker")]:</A>
-					</td>
-					<td class="column2">
-						<A href='?src=\ref[src];removeall=1;percent=[100/3]'>33%</A>
-						<A href='?src=\ref[src];removeall=1;percent=50'>50%</A>
-						<A href='?src=\ref[src];removeallcustom=1'>Custom</A>
-						<A href='?src=\ref[src];removeall=1;amount=[reagents.total_volume]'>All</A>
-					</td>
-				</table>
-			"}
-
-			dat += "<table>"
-			for(var/datum/reagent/N in reagents.reagent_list)
-				dat += "<tr>"
-				dat += {"
-					<td class="column1">
-						[N.name] , [round(N.volume, 0.01)] Units - <A href='?src=\ref[src];analyze=\ref[N]'>(?)</A>
-					</td>
-					<td class="column2">
-						<A href='?src=\ref[src];remove=[N.id];amount=1'>1u</A>
-						<A href='?src=\ref[src];remove=[N.id];amount=5'>5u</A>
-						<A href='?src=\ref[src];remove=[N.id];amount=10'>10u</A>
-						<A href='?src=\ref[src];removecustom=[N.id]'>Custom</A>
-						<A href='?src=\ref[src];remove=[N.id];amount=[N.volume]'>All</A>
-					</td>
-				"}
-				dat += "</tr>"
-			dat += "</table>"
-		else
-			dat += "Buffer is empty.<BR>"
+				if(loaded_pill_bottle.contents.len < loaded_pill_bottle.storage_slots)
+					P.forceMove(loaded_pill_bottle)
+			if(count == 0) //only do this ONCE
+				logged_message += "[P.reagents.get_reagent_ids(1)]. Icon: [pillIcon2Name[text2num(pillsprite)]]"
 
-		if(condi)
-			dat += "<A href='?src=\ref[src];createbottle=1'>Create bottle (50 units max)</A>"
-		else
-			//
-			// PILL ICONS
-			//
-			dat += "<HR>"
-			dat += "<div class='pillIconsContainer'>"
-			for(var/i = 1 to MAX_PILL_SPRITE)
-				dat += {"<a href="?src=\ref[src]&pill_sprite=[i]" class="pillIconWrapper[i == text2num(pillsprite) ? " linkOnMinimal" : ""]">
-							<div class="pillIcon">
-								[pill_icon_cache[i]]
-							</div>
-						</a>"}
-				if (i%10 == 0)
-					dat +="<br>"
-			dat += "</div>"
-			//
-			// BUTTONS
-			//
-			dat += {"<HR><A href='?src=\ref[src];createpill=1'>Create single pill ([max_pill_size] units max)</A><BR>
-					<A href='?src=\ref[src];createpill_multiple=1'>Create multiple pills ([max_pill_size] units max each; [max_pill_count] max)</A><BR>
-					<A href='?src=\ref[src];createpill_multiple=1;createempty=1'>Create empty pills</A><BR>
```

Comment:
bring this back, no reason to remove this, especially on the old chem master that you shouldn't be changing anyway